pkg/cloudprovider/aws/fleet/vpc.go

```diff
@@ -0,0 +1,158 @@
+/*
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package fleet
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/ec2"
+	"github.com/aws/aws-sdk-go/service/ec2/ec2iface"
+	"github.com/awslabs/karpenter/pkg/apis/provisioning/v1alpha1"
+	"github.com/awslabs/karpenter/pkg/cloudprovider"
+	"github.com/patrickmn/go-cache"
+	"go.uber.org/zap"
+)
+
+type VPCProvider struct {
+	launchTemplateProvider *LaunchTemplateProvider
+	subnetProvider         *SubnetProvider
+}
+
+func (p *VPCProvider) GetLaunchTemplate(ctx context.Context, clusterSpec *v1alpha1.ClusterSpec) (*ec2.LaunchTemplate, error) {
```

Comment:
What's the value of wrapping the launchTemplateProvider inside the vpc concept?

Reply:
I wanted a single abstraction over the things we need to configure/check like launchTemplate, subnet for creating an instance. Later may be we will add things like security groups or SSM they can all go under this abstraction and we can just pass the `vpcProvider` around.

Reply:
Ok -- this approach makes sense to me. I'm not sure VPC is the right name though -- sorry to nitpick so much about names. In some sense, it is more or less the role of this entire package (i.e. dealing with subnets, security groups, zones, launch templates, etc). The size of that scope is what originally led me to break these apart into separate files and tie them together with capacity.go. 

I expect all of this will evolve over time, and don't want to spend too much time bike shedding. I think it's best to just go with what you think is best (unless I see something obviously wrong/dangerous), and we can expect to keep reshaping the code as we grow.